src/Backend/Modules/Location/Engine/Model.php

```diff
@@ -2,88 +2,63 @@
 
 namespace Backend\Modules\Location\Engine;
 
-use Backend\Core\Language\Language as BL;
+use Backend\Core\Language\Language;
 use Backend\Core\Engine\Model as BackendModel;
+use Backend\Modules\Location\Domain\Location;
+use Backend\Modules\Location\Domain\LocationSetting;
 use Common\ModuleExtraType;
-use Symfony\Component\Intl\Intl as Intl;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityRepository;
+use InvalidArgumentException;
+use SpoonFilter;
+use Symfony\Component\Intl\Intl;
 
-/**
- * In this file we store all generic functions that we will be using in the location module
- */
 class Model
 {
     const QUERY_DATAGRID_BROWSE =
         'SELECT id, title, CONCAT(street, " ", number, ", ", zip, " ", city, ", ", country) AS address
          FROM location
-         WHERE language = ?';
+         WHERE locale = ?';
 
-    /**
-     * Delete an item
-     *
-     * @param int $id The id of the record to delete.
-     */
     public static function delete(int $id): void
     {
-        // get database
-        $database = BackendModel::getContainer()->get('database');
+        $location = self::getLocationRepository()->find($id);
 
-        // get item
-        $item = self::get($id);
+        if ($location instanceof Location) {
+            BackendModel::deleteExtraById($location->getExtraId());
 
-        BackendModel::deleteExtraById($item['extra_id']);
+            $entityManager = self::getEntityManager();
 
-        // delete location and its settings
-        $database->delete('location', 'id = ? AND language = ?', [$id, BL::getWorkingLanguage()]);
-        $database->delete('location_settings', 'map_id = ?', [$id]);
+            $entityManager->remove($location);
+            $entityManager->flush();
```

Comment:
is there a reason you are not doing `$entityManager->flush($location);` here?

Reply:
9482a995c